include/ua_plugin_securitypolicy.h

```diff
@@ -0,0 +1,424 @@
+    */
+    size_t (*const getLocalAsymSignatureSize)(const UA_SecurityPolicy *securityPolicy,
+                                              const void *endpointContext);
+
+    /**
+     * \brief Compares the supplied certificate with the certificate in the endpoit context.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param endpointContext the endpoint context data that contains the certificate to compare to.
+     * \param certificateThumbprint the certificate thumbprint to compare to the one stored in the context.
+     * \return if the thumbprints match UA_STATUSCODE_GOOD is returned. If they don't match
+     *         or an errror occured an error code is returned.
+     */
+    UA_StatusCode (*const compareCertificateThumbprint)(const UA_SecurityPolicy *securityPolicy,
+                                                        const void *endpointContext,
+                                                        const UA_ByteString *certificateThumbprint);
+};
+
+struct UA_Channel_SecurityContext {
+    /**
+     * \brief This method initializes a new context data object.
+     * The caller needs to call deleteMembers on the recieved object to free allocated memory.
+     *
+     * Memory is only allocated if the function succeeds so there is no need to manually free
+     * the memory pointed to by *pp_contextData or to call deleteMembers.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param remoteCertificate the remote certificate contains the remote asymmetric key.
+     *                          The certificate will be verified and then stored in the context
+     *                          so that its details may be accessed.
+     * \param contextData the initialized contextData that is passed to functions that work
+     *                    on a context.
+     */
+    UA_StatusCode (*const init)(const UA_SecurityPolicy *securityPolicy,
+                                const void *endpointContext,
+                                const UA_ByteString *remoteCertificate,
+                                void **pp_contextData);
+
+    /**
+     * \brief Deletes the members of the security context.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param contextData the context to delete the members of.
+     */
+    UA_StatusCode (*const deleteMembers)(const UA_SecurityPolicy *securityPolicy,
+                                         void *contextData);
+
+    /**
+     * \brief Sets the local encrypting key in the supplied context.
+     *
+     * \param securityPolicy contains the function pointers associated with the policy.
+     * \param key the local encrypting key to store in the context.
+     * \param contextData the context to work on.
+     */
+    UA_StatusCode (*const setLocalSymEncryptingKey)(const UA_SecurityPolicy *securityPolicy,

```

Comment:
That should be done by the _new (_init) method in the background.
There server does not have this key, anyway.

So it is not part of the API that interfaces between the securitypolicy and the server at runtime.